src/main/java/ch/njol/skript/effects/EffEquip.java

```diff
@@ -164,9 +171,12 @@ protected void execute(Event event) {
 							equipment.setLeggings(equip ? item : null);
 						} else if (BOOTS.isOfType(item)) {
 							equipment.setBoots(equip ? item : null);
-						} else {
-							// Apply all other items to head, as all items will appear on a player's head
+						} else if (HELMET.isOfType(item) || isHat) {
+							// Apply all other items to head (if isHat), as all items will appear on a player's head
 							equipment.setHelmet(equip ? item : null);
+						} else {
+							if (entity instanceof Player)
+								((Player) entity).getInventory().addItem(item);
```

Comment:
I don't believe we should just add the item, since we're equipping it should we not set the main hand?

Reply:
Well, 'equip' can actually mean giving items so I don't see a problem with that but I do see that setting main hand is also good option however, it's becoming weirder syntax because we have to add extra syntax parts in order to know what the user wants, if you could suggest a syntax that might help.

Reply:
To be honestly the easiest way to equip anything would be to use bukkit's own EquipmentSlot enum which uses `FEET, LEGS, CHEST, HEAD, OFFHAND, MAINHAND` while this means changes to EquipmentSlot in skript it's better to use this reguardless for entities
```java
EntityEquipment entityEquipment = instance.getEntityEquipment()
entityEquipment.setItem(equipmentSlot, item
```

syntax wise it'll just be about this, excluding the unequip syntax
```applescript
equip %equipment slot% of %livingentities% with %itemtype%
equip [%livingentities%'[s]] %equipment slot% with %itemtype%
equip [%livingentities%] with %itemtype%
```
```
equip player's chestplate with stone
unequip player's helmet 
```

Reply:
It's becoming more like `set player's chestplate to diamond chestplate` while `equip` is supposed to be quicker to use and easier `equip player with diamond chestplate`, so I think we are moving away from the main purpose here.

For this PR, I will keep it how it is and maybe later on we can improve it more if we found a good way to.